features/priority_failover.feature

```diff
@@ -0,0 +1,23 @@
+Feature: priority replication
+  We should check that we can give nodes priority during failover
+
+  Scenario: check failover priority 0 prevents leaderships
+    Given I configure and start postgres0 with a tag failover_priority 1
+    And I configure and start postgres1 with a tag failover_priority 0
+    Then replication works from postgres0 to postgres1 after 20 seconds
+    When I shut down postgres0
+    And I sleep for 5 seconds
```

Comment:
Such sleep calls are not very reliable. Sometimes we get so slow MacOS instances that tests are still failing.
In general it is better to check some condition in a loop instead.
For example:
```cucumber
    And there is a "following a different leader because I am not allowed to promote" INFO in the postgres1 patroni log after 5 seconds
```
And perform the check every second in a loop (example): https://github.com/zalando/patroni/blob/3333e78500204159332e032e5d07727dfbb7129d/features/steps/patroni_api.py#L136-L147

That is., fail on timeout rather than on a single attempt after sleep. The advantage is that on success we interrupt the loop much earlier. Also, there is a possibility on increase the timeout when it is known that tests are executed on CI: https://github.com/zalando/patroni/blob/3333e78500204159332e032e5d07727dfbb7129d/features/environment.py#L1060-L1062

Reply:
This makes a lot of sense, thanks for the in-depth explanation. Should I rewrite parts of this test to avoid the sleep here?

Reply:
@mfpekala-nutanix yes, please rewrite them using to use 1s polling with configurable timeout before fail instead of using sleep.